packages/renderer/src/stores/kubernetes-context-health.ts

```diff
@@ -25,8 +25,18 @@ import { EventStore } from './event-store';
 const windowEvents = ['kubernetes-contexts-healths', 'extension-stopped', 'extensions-started'];
 const windowListeners = ['extensions-already-started'];
 
+let experimentalStates: boolean | undefined = undefined;
 let readyToUpdate = false;
+
 export async function checkForUpdate(eventName: string): Promise<boolean> {
+  // check for update only in experimental states mode
+  if (experimentalStates === undefined) {
+    experimentalStates = (await window.getConfigurationValue<boolean>('kubernetes.statesExperimental')) ?? false;
+  }
+  if (experimentalStates === false) {
+    return false;
+  }
+
```

Comment:
hello, is it ok to check only once the configuration ? if for example I'm turning on the experimental flag later it won't be handled ?

should it monitor the value of this property ?

Reply:
The live change is not supported elsewhere in the code, the user needs to restart the app (as the config is hidden, it does not seem necessary to me, and it would complexify te `kubernetes-client` logic).